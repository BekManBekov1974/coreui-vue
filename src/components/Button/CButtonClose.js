import { mergeData } from 'vue-functional-data-merge'
export default {
  name: 'CButtonClose',
  functional: true,
  props: {
    iconHtml: {
      type: String,
      default: '&times;'
    }
  },
  render (h, { data, props, slots }) {
    return h(
      'button',
      mergeData(data,  {
        staticClass: 'close',
        attrs: {
          type: props.type || 'button',
          'aria-label': props.ariaLabel || 'Close'
        },
        domProps: !slots().default ? { innerHTML: props.iconHtml } : null
      }),
      slots().default
    )
  }
}
