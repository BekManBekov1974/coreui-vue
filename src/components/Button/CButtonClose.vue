<script>
import { mergeData } from 'vue-functional-data-merge'
export default {
  name: 'CButtonClose',
  functional: true,
  props: {
    iconHtml: {
      type: String,
      default: '&times;'
    },
    buttonClasses: {
      type: [String, Array, Object],
      default: 'close'
    }
  },
  render (h, { data, props, slots }) {
    return h(
      'button',
      mergeData(data,  {
        class: props.buttonClasses,
        attrs: {
          type: 'button',
          'aria-label': 'Close'
        },
        domProps: !slots().default ? { innerHTML: props.iconHtml } : null
      }),
      slots().default
    )
  }
}
</script>

<style lang="scss">
  @import "~@coreui/coreui/scss/partials/buttons.scss";
  @import "~@coreui/coreui/scss/partials/close.scss";
</style>
