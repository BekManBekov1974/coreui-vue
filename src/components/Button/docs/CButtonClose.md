### Button close (CButtonClose) [Functional]

Simple closing button component.

prop | default | valid values
--- | --- | ---
iconHtml  | `'&times;'` | String
buttonClasses  | `'close'` | String

You can add standard html attributes freely, as you would do in normal button element,
they will overwrite default behaviour. By default component will output:

```html
<button
  aria-label="Close"
  class="close"
  type="button"
>
  &times;
</button>
```

If component have default slot, it will overwrite iconHtml prop.
