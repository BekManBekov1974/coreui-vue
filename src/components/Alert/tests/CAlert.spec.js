import { mount } from '@vue/test-utils'
import Component from "../CAlert";
import CButtonClose from "../../Button/CButtonClose";
import Vue from 'vue'

Vue.component('CButtonClose', CButtonClose)

const ComponentName = 'CAlert'
const defaultWrapper = mount(Component, {
  propsData: {
    show: true,
    dismissible: true
  },
  slots: {
    default: 'Dismissible Alert!'
  }
})

const customWrapper = mount(Component, {
  propsData: {
    show: true,
    fade: true,
    dismissible: true,
    variant: 'success'
  },
  slots: {
    default: 'Dismissible Alert!'
  }
})

describe(ComponentName, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('renders correctly', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
  it('renders correctly', () => {
    expect(customWrapper.element).toMatchSnapshot()
  })
})
