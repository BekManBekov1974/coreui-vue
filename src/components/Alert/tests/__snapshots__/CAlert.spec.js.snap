// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CAlert renders correctly 1`] = `
<div
  appear="true"
  aria-atomic="true"
  aria-live="polite"
  class="c-alert c-alert-dismissible c-alert-primary"
  role="alert"
>
  <button
    aria-label="Close"
    class="c-close"
    type="button"
  >
    ×
  </button>
   Dismissible Alert!
</div>
`;

exports[`CAlert renders correctly 2`] = `
<div
  appear="true"
  aria-atomic="true"
  aria-live="polite"
  class="c-alert c-alert-dismissible c-alert-success"
  name="fade"
  role="alert"
>
  <button
    aria-label="Close"
    class="c-close"
    type="button"
  >
    ×
  </button>
   Dismissible Alert!
</div>
`;
